Tools/Scripts/libraries/webkitcorepy/webkitcorepy/string_utils.py

```diff
@@ -92,23 +92,31 @@ def out_of(number, base):
     return '[{}{}/{}]'.format(' ' * (len(base) - len(number)), number, base)
 
 
-def elapsed(seconds):
-    if seconds <= 0:
-        return 'no time'
-    elif seconds < 1:
-        return 'less than a second'
-
-    seconds = int(round(seconds))
+def elapsed(seconds: float, precision: int = None, shorthand: bool = False):
+    rounded_seconds = round(seconds, ndigits=precision)
+    if seconds < 60:
+        if shorthand:
+            return f'{rounded_seconds}s' if not (rounded_seconds == 0 and seconds > 0) else '<1s'
+        elif precision is None:
+            if seconds <= 0:
+                return 'no time'
+            elif seconds < 1:
+                return 'less than a second'
+        return pluralize(rounded_seconds, 'second')
+
+    seconds = int(seconds)
     minutes = seconds // 60
     hours = minutes // 60
     days = hours // 24
 
+    precise_seconds = (seconds % 60) + round(rounded_seconds - seconds, ndigits=precision)
```

Comment:
What is this trying to calculate? The number of seconds?

Because in the `59.9` case currently, this will be equal to `60`, when presumably we're expecting a value in the range [0,60) (i.e., `0.0 ≤ precise_seconds < 60.0`).